### ISSUE TITLE
Replace text substitution method of passing arguments into Python operators

### ISSUE THREAD

**Issue by cquammen** [OpenChemistry/tomviz#704]
Alternative implementation possibilities:
- Make all Operators a subclass of the `tomviz.operators.Operator` and provide a member dictionary containing the arguments
- Add a second argument to `transform_scalars` functions that is a dictionary containing arguments
- Global variable that is defined only in the Python environment in which the operator python operates.

Instead of a dictionary, individual Python variables/members could be defined within an appropriate scope.

Made more pressing by issues reported in #702.

**Comment by cjh1** [comment]
@cquammen For option one my idea was the actually add the argument as members of the object for example if you had an argument foo, it could be be access inside the operator as ```self.foo```

**Comment by cquammen** [comment]
@cjh1 I see, I've updated the original comment to say 'individual Python variables/members could be defined...'. One thing to be careful about with that approach is that there will be some, maybe not many, reserved names that cannot be used as argument names. Also, every Python Operator will have to be updated to be a `tomviz.operators.Operator` subclass. Or maybe there is a fancy way to make top-level `transform_scalars` functions member functions of a generic Operator subclass, but that might get confusing when referring to `self.member` inside a plain Python function.

I think that no matter what we do, arguments should be namespaced in some way, e.g., `self.args['foo']`, `self.settings['foo']`, `self.settings.foo`, or something to that effect.

**Comment by cjh1** [comment]
@cquammen I had thought about the possibility of `name clash` using this approach, both user defined members and system ones, however we could easily warn the user. I think we are going to have to touch every operator anyway if we are moving away from templating the variables, so converting every operator to subclass Operator is not a big deal. However, I am warming to the idea for changing the ```transform_scalars``` to something like ```transform_scalars(dataset, **kwargs)```. May be we should discuss this at the next meeting?

**Comment by cryos** [comment]
Chris and I chatted about this on Friday a little, members or argument
object both seem reasonable. Short term, flake8 should be relaxed or turned
off so we can continue without the urgency

**Comment by cjh1** [comment]
I have started to prototype pass arguments to the ```transform_scalars``, so for example the signature in GaussianFilter.py changes from:

```python
def transform_scalars(dataset);
    ...
```

to 

```
def transform_scalars(dataset, sigma=None):
   ...
```

My plan was to get it working with a simple operator and see how it works/what it looks like.

**Comment by cjh1** [comment]
Fixed in #766
